core/blockchain/blockchain_gateway.py

```diff
@@ -0,0 +1,355 @@
+import asyncio
+import base58
+import json
+import logging
+import os
+import requests
+import time
+
+import ipfsapi
+
+from core.configuration import ConfigurationManager
+from core.utils.event_types import MessageEventTypes
+
+
+logging.basicConfig(level=logging.DEBUG,
+    format='[BlockchainGateway] %(message)s')
+
+
+class BlockchainGateway(object):
+    """
+    Blockchain Gateway 
+
+    The blockchain client exposes `setter` and `getter` in order to interact
+    with the blockchain.
+
+    In order for this to work, the following must be running:
+        IPFS Daemon: `ipfs daemon`
+        The lotion app: `node app_trivial.js` from the application root directory
+
```

Comment:
There's nothing explaining how listening works/is operated in this class.

Reply:
How in depth should we have docstrings be? they're already adding a lot of bulk to an otherwise lean file

Reply:
It's okay to add more lines if they help with readability of the code. Explain enough s.t. a new developer in the team doesn't need to dig into the code to know what this is doing.